### ISSUE TITLE
building fabric images for arm64 architecture

### ISSUE THREAD

**Issue by rohitroyrr8** [hyperledger/fabric#2895]
I am looking for how can I build docker images for fabric components like peer, orderers, ccenv etc for arm based architecture.  I pulled fabric, fabric-ca and fabric-baseimage and make changes in respected Dockerfile but while running make getting below error. Please help me in that.

```
Docker: building couchdb image
docker build   -t hyperledger/fabric-couchdb -f images/couchdb/Dockerfile images/couchdb
[+] Building 6.8s (11/17)                                                                                                
 => [internal] load build definition from Dockerfile                                                                0.0s
 => => transferring dockerfile: 37B                                                                                 0.0s
 => [internal] load .dockerignore                                                                                   0.0s
 => => transferring context: 2B                                                                                     0.0s
 => [internal] load metadata for docker.io/arm64v8/debian:buster                                                    3.4s
 => [auth] arm64v8/debian:pull token for registry-1.docker.io                                                       0.0s
 => [internal] load build context                                                                                   0.0s
 => => transferring context: 151B                                                                                   0.0s
 => [ 1/12] FROM docker.io/arm64v8/debian:buster@sha256:b4cf9a9f5266cd9e75a5648238831b88ed4d124567f2e3a5beca0d9dab  0.0s
 => CACHED [ 2/12] RUN groupadd -g 5984 -r couchdb && useradd -u 5984 -d /opt/couchdb -g couchdb couchdb            0.0s
 => CACHED [ 3/12] RUN apt-get update -y && apt-get install -y --no-install-recommends     ca-certificates     cur  0.0s
 => CACHED [ 4/12] RUN set -ex;   apt-get update;  apt-get install -y --no-install-recommends wget;  rm -rf /var/l  0.0s
 => CACHED [ 5/12] RUN set -xe     && for key in 15DD4F3B8AACA54740EB78C7B7B7C53943ECCEE1   1CFBFA43C19B6DF4A0CA39  0.0s
 => ERROR [ 6/12] RUN buildDeps='     apt-transport-https     gcc     g++     erlang-dev     libcurl4-openssl-dev   3.4s
------                                                                                                                   
 > [ 6/12] RUN buildDeps='     apt-transport-https     gcc     g++     erlang-dev     libcurl4-openssl-dev     libicu-dev     libmozjs185-dev     make   '  && apt-get update -y -qq && apt-get install -y --no-install-recommends $buildDeps  && cd /usr/src && mkdir couchdb  && curl -fSL https://archive.apache.org/dist/couchdb/source/2.3.1/apache-couchdb-2.3.1.tar.gz -o couchdb.tar.gz  && curl -fSL https://archive.apache.org/dist/couchdb/source/2.3.1/apache-couchdb-2.3.1.tar.gz.asc -o couchdb.tar.gz.asc  && gpg --batch --verify couchdb.tar.gz.asc couchdb.tar.gz  && tar -xzf couchdb.tar.gz -C couchdb --strip-components=1  && cd couchdb  && ./configure --disable-docs  && make release  && mv /usr/src/couchdb/rel/couchdb /opt/  && apt-get purge -y --auto-remove $buildDeps  && rm -rf /var/lib/apt/lists/* /usr/src/couchdb*  && mkdir /opt/couchdb/data  && chown -R couchdb:couchdb /opt/couchdb:
#10 2.919 Reading package lists...
#10 3.242 Building dependency tree...
#10 3.303 Reading state information...
#10 3.346 E: Unable to locate package libmozjs185-dev
------
executor failed running [/bin/sh -c buildDeps='     apt-transport-https     gcc     g++     erlang-dev     libcurl4-openssl-dev     libicu-dev     libmozjs185-dev     make   '  && apt-get update -y -qq && apt-get install -y --no-install-recommends $buildDeps  && cd /usr/src && mkdir couchdb  && curl -fSL https://archive.apache.org/dist/couchdb/source/$COUCHDB_VERSION/apache-couchdb-$COUCHDB_VERSION.tar.gz -o couchdb.tar.gz  && curl -fSL https://archive.apache.org/dist/couchdb/source/$COUCHDB_VERSION/apache-couchdb-$COUCHDB_VERSION.tar.gz.asc -o couchdb.tar.gz.asc  && gpg --batch --verify couchdb.tar.gz.asc couchdb.tar.gz  && tar -xzf couchdb.tar.gz -C couchdb --strip-components=1  && cd couchdb  && ./configure --disable-docs  && make release  && mv /usr/src/couchdb/rel/couchdb /opt/  && apt-get purge -y --auto-remove $buildDeps  && rm -rf /var/lib/apt/lists/* /usr/src/couchdb*  && mkdir /opt/couchdb/data  && chown -R couchdb:couchdb /opt/couchdb]: exit code: 100
make: *** [build/image/couchdb/.arm64-0.4.20] Error 1
```

**Comment by mastersingh24** [comment]
@rohitroyrr8  - which version are you building?  We no longer build couchdb images and we no longer build anything from fabric-baseimage except for critical fixes for Fabric 1.4.x which is not really supported.

CouchDB does provide an arm64 image now themselves as well, so the Makefile in Fabric main should pull the arm64 version

**Comment by georgejdli** [comment]
Is there a way to build the 1.4 thirdparty images (couchdb) for arm64?
I'm looking at the makefile for release-1.4 and it looks like it only pulls the thirdparty images from dockerhub
Everything else seems to work in amd64 emulation but the couchdb container crashes

Edit:
So instead of using the fabric built docker image for couchdb you're saying we can use the official version instead?
docker pull couchdb:2.3.1 
(couchdb v2 for fabric v1.4)
